src/proof/EcdsaSecp256k1RecoverySignature2020.ts

```diff
@@ -0,0 +1,286 @@
+//     controller: 'did:key:zQ3shjRPgHQQbTtXyofk1ygghRJ75RZpXmWBMY1BKnhyz7zKp',
+//     publicKeyJwk: {
+//         kty: 'EC',
+//         crv: 'secp256k1',
+//         x: 'RwiZITTa2Dcmq-V1j-5tgPUshOLO31FbsnhVS-7lskc',
+//         y: '3o1-UCc3ABh757P58gDISSc4hOj9qyfSGl3SGGA7xdc'
+//     },
+//     privateKeyJwk: {
+//         kty: 'EC',
+//         crv: 'secp256k1',
+//         x: 'RwiZITTa2Dcmq-V1j-5tgPUshOLO31FbsnhVS-7lskc',
+//         y: '3o1-UCc3ABh757P58gDISSc4hOj9qyfSGl3SGGA7xdc',
+//         d: 'T2azVap7CYD_kB8ilbnFYqwwYb5N-GcD6yjGEvquZXg'
+//     }
+// }
+
+type ProofFormat = "vc" | "vc-jwt";
+
+export default class EcdsaSecp256k1RecoverySignature2020Proof implements Proof<any> {
+    constructor(private resolver: IDiDResolver, private format: ProofFormat = "vc") {
+    }
+
+    async sign(credential: VerifiableCredential, key: any): Promise<VerifiableCredential> {
+        credential["@context"].push("https://w3id.org/security/suites/secp256k1recovery-2020/v2");
+
+        const result = await verifiable.credential.create({
+            credential,
+            format: [this.format],
+            documentLoader: this.documentLoader.bind(this),
+            suite: new EcdsaSecp256k1RecoverySignature2020({
+                key: await EcdsaSecp256k1RecoveryMethod2020.from(key),
+                date: credential.issuanceDate,
+            }),
+        });
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return result.items[0];
+    }
+
+    async verify(credential: VerifiableCredential): Promise<boolean> {
+        const result = await verifiable.credential.verify({
+            credential,
+            format: [this.format],
+            documentLoader: this.documentLoader.bind(this),
+            suite: new EcdsaSecp256k1RecoverySignature2020({
+                date: credential.issuanceDate,
+            }),
+        });
+
+        return result.verified;
+    }
+
+    private async documentLoader(iri: string) {
+        console.log("IRI: " + iri);
+
+        if (contexts[iri]) {
+            return {document: contexts[iri]};
+        }
+
+        if (/^did:/.test(iri)) {
+            return this.createController(iri);
+        }
+
+        throw new Error(`Unsupported iri: ${iri}`);
+    }
+
+    private async createController(iri: string) {
+        try {
+            console.log("CREATE CONTOLLER A");
+
+            const did = iri.split("#")[0];
+            console.log("CREATE CONTOLLER B");
+
+            const doc = await this.resolver.resolve(did);
+            console.log("CREATE CONTOLLER C");
+
+            if (!doc) {
+                throw new Error(`No document found for  ${did}`)
+            }
+
+            const foundKey = doc.verificationMethod?.find((pk: any) => pk.id.startsWith(iri));
+            console.log("CREATE CONTOLLER D");
+            console.log(foundKey);
+
+            const key = await EcdsaSecp256k1RecoveryMethod2020.from(foundKey);
```

Comment:
The key on the did is of type `EcdsaSecp256k1RecoverySignature2019`... I believe fixing this will resolve the problem

Reply:
@rado0x54 this is my main problem